### PR TITLE
Fix renderWindow's size not follow outputDelegate to change

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: src/server/kernel/woutputlayout.cpp [hunk 1]
```diff
@@ -17,6 +17,29 @@ WOutputLayoutPrivate::WOutputLayoutPrivate(WOutputLayout *qq)
 
 }
 
+void WOutputLayoutPrivate::updateImplicitSize()
+{
+    W_Q(WOutputLayout);
+
+    int newWidth = 0, newHeight = 0;
+    for (auto outputPtr : outputs)
```
Comment (zccrs) [inline src/server/kernel/woutputlayout.cpp:25]: You can use `QWOutputLayouer::getBox(nullptr)`

File: src/server/kernel/woutputlayout.cpp [hunk 2]
```diff
@@ -57,7 +85,17 @@ void WOutputLayout::move(WOutput *output, const QPoint &pos)
     if (output->position() == pos)
         return;
 
+    bool maybeImplicitSizeChange = false;
```
Comment (zccrs) [inline src/server/kernel/woutputlayout.cpp:88]: Why not direct call `updateImplicitSize`

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/platformplugin/qwlrootsintegration.cpp [hunk 3]
```diff
@@ -226,15 +226,14 @@ QInputDevice *QWlrootsIntegration::addInputDevice(WInputDevice *device, const QS
     return qtdev;
 }
 
-QInputDevice *QWlrootsIntegration::removeInputDevice(WInputDevice *device)
+bool QWlrootsIntegration::removeInputDevice(WInputDevice *device)
 {
     if (auto qdevice = getInputDeviceFrom(device)) {
-        device->setQtDevice(nullptr);
```
[the remark this answers is not in the document]
Comment (wineee) [reply src/server/platformplugin/qwlrootsintegration.cpp:232]: device use QPinter now

File: src/server/kernel/woutputlayout.cpp (hunk 2 above)
Comment (wineee) [reply src/server/kernel/woutputlayout.cpp]: not necessary

File: src/server/qtquick/woutputlayoutitem.h [hunk 4]
```diff
@@ -26,12 +26,12 @@ class WAYLIB_SERVER_EXPORT WOutputLayoutItem : public WQuickObserver, public WOb
     Q_OBJECT
     QML_NAMED_ELEMENT(OutputLayoutItem)
     Q_PROPERTY(WQuickOutputLayout* layout READ layout WRITE setLayout NOTIFY layoutChanged)
-    Q_PROPERTY(QList<WOutput*> outputs READ outputs RESET resetOutput NOTIFY outputsChanged)
+    Q_PROPERTY(QList<QPointer<WOutput>> outputs READ outputs RESET resetOutput NOTIFY outputsChanged)
```
Comment (zccrs) [inline src/server/qtquick/woutputlayoutitem.h:29]: Can using by QML

Comment (wineee) [reply src/server/qtquick/woutputlayoutitem.h]: No, can,t

File: src/server/kernel/woutputlayout.cpp [hunk 5]
```diff
@@ -68,15 +91,17 @@ void WOutputLayout::remove(WOutput *output)
...
     QWOutputLayout::remove(output->handle());
     output->setLayout(nullptr);
+    output->disconnect(this);
+    d->updateImplicitSize();
 
     Q_EMIT outputRemoved(output);
 }
 
-QList<WOutput *> WOutputLayout::getIntersectedOutputs(const QRect &geometry) const
+QList<QPointer<WOutput>> WOutputLayout::getIntersectedOutputs(const QRect &geometry) const
```
Comment (zccrs) [inline src/server/kernel/woutputlayout.cpp:100]: Keep return value type

File: src/server/qtquick/woutputlayoutitem.cpp [hunk 6]
```diff
@@ -87,10 +88,14 @@ void WOutputLayoutItem::resetOutput()
     setLayout(nullptr);
 }
 
-QList<WOutput*> WOutputLayoutItem::outputs() const
+QList<WOutput *> WOutputLayoutItem::outputs() const
```
Comment (zccrs) [inline src/server/qtquick/woutputlayoutitem.cpp:91]: No space

File: src/server/qtquick/woutputlayoutitem.h [hunk 7]
```diff
@@ -26,12 +26,13 @@ class WAYLIB_SERVER_EXPORT WOutputLayoutItem : public WQuickObserver, public WOb
     Q_OBJECT
     QML_NAMED_ELEMENT(OutputLayoutItem)
     Q_PROPERTY(WQuickOutputLayout* layout READ layout WRITE setLayout NOTIFY layoutChanged)
-    Q_PROPERTY(QList<WOutput*> outputs READ outputs RESET resetOutput NOTIFY outputsChanged)
+    Q_PROPERTY(QList<WOutput *> outputs READ outputs RESET resetOutput NOTIFY outputsChanged)
```
Comment (zccrs) [inline src/server/qtquick/woutputlayoutitem.h:29]: No space

File: src/server/qtquick/woutputlayoutitem.h [hunk 8]
```diff
@@ -26,12 +26,13 @@ class WAYLIB_SERVER_EXPORT WOutputLayoutItem : public WQuickObserver, public WOb
...
 
 public:
     explicit WOutputLayoutItem(QQuickItem *parent = nullptr);
 
-    QList<WOutput*> outputs() const;
+    // Can't use QPointer in QML
```
Comment (zccrs) [inline src/server/qtquick/woutputlayoutitem.h:34]: Not needs comments

File: src/server/kernel/woutputlayout.cpp [hunk 9]
```diff
@@ -30,7 +46,7 @@ WOutputLayout::WOutputLayout(QObject *parent)
 
 }
 
-QList<WOutput*> WOutputLayout::outputs() const
+QList<QPointer<WOutput>> WOutputLayout::outputs() const
```
Comment (zccrs) [inline src/server/kernel/woutputlayout.cpp:49]: Keep QList<WOutput*>

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**